internals/overlord/overlord.go

```diff
@@ -321,6 +328,25 @@ func (o *Overlord) ensureTimerSetup() {
 	o.pruneTicker = time.NewTicker(pruneInterval)
 }
 
+func (o *Overlord) ensureTimerStopClearChannel() {
+	o.ensureLock.Lock()
+	defer o.ensureLock.Unlock()
+	// The Go v1.x timer implementation does not guarantee that a Reset
```

Comment:
Interestingly, this is being fixed in Go 1.23, which is coming out soon. https://tip.golang.org/doc/go1.23#timer-changes

Reply:
We may have to check with snapd to see which versions they have to support. I think they are slightly less flexible due to the range of distros they have to build on.

Reply:
Yeah, I'm not suggesting a change here, it was just a note.